### PR TITLE
Update to Java 17

### PR DESCRIPTION
**Pull request by SteveScorfield**
# What and why?
We need to update the version of Java to version 17 along with other related dependencies
# How to test?
Run against unit and acceptance tests.
# Trello
https://trello.com/c/r9DKr9pB/1625-upgrade-rm-collection-exercise-to-java-17

### CODE REVIEW

**Comment by SteveScorfield** [comment]
[Upgrade rm-collection-exercise to java 17](https://trello.com/c/r9DKr9pB/1625-upgrade-rm-collection-exercise-to-java-17)

File: pom.xml [hunk 1]
```diff
@@ -212,12 +200,13 @@
 
         <!-- Testing -->
 
-        <!-- Added because of mockito errors. It's explained more here
-        https://dev.to/scottshipp/how-to-fix-a-mockito-cannot-mock-this-class-exception-in-a-spring-boot-app-457e -->
+        <!-- Added because of mockito errors. mockito-core was causing null pointer exceptiosn. This is caused by the
```
Comment (insacuri) [inline pom.xml:203]: Small spelling mistake in exceptions

File: pom.xml [hunk 2]
```diff
@@ -153,6 +140,7 @@
         <dependency>
             <groupId>org.liquibase</groupId>
             <artifactId>liquibase-core</artifactId>
+            <version>4.11.0</version>
```
Comment (insacuri) [inline pom.xml:143]: Why does there need to be a version attached to this now?

File: src/main/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/ObjectConverter.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package uk.gov.ons.ctp.response.collection.exercise.endpoint;
```
Comment (insacuri) [inline src/main/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/ObjectConverter.java:1]: Having this be in the endpoint folder feels wrong as this doesn't deal with http requests.  Maybe in domain or representation?

File: src/test/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpointUnitTests.java [hunk 4]
```diff
@@ -100,7 +98,7 @@ public class CollectionExerciseEndpointUnitTests {
 
   @Mock private EventService eventService;
 
-  @Spy private MapperFacade mapperFacade = new CollectionExerciseBeanMapper();
+  //  @Spy private MapperFacade mapperFacade = new CollectionExerciseBeanMapper();
```
Comment (insacuri) [inline src/test/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpointUnitTests.java:101]: Delete this if it's not needed

File: pom.xml (hunk 1 above)
Comment (SteveScorfield) [reply pom.xml:203]: Corrected

**Review by insacuri — CHANGES_REQUESTED** [review]
Looks good, just a few small bits to sort out

File: src/main/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/ObjectConverter.java (hunk 3 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/ObjectConverter.java:1]: You are right. Thinking about it now it shouldn't be here. I've moved it to domain as suggested. It's more relevant.

File: src/test/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpointIT.java [hunk 5]
```diff
@@ -88,10 +87,6 @@ public class CollectionExerciseEndpointIT {
 
   @Rule public final SpringMethodRule springMethodRule = new SpringMethodRule();
 
-  @ClassRule
```
[the remark this answers is not in the document]
Comment (SteveScorfield) [reply src/test/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpointIT.java:91]: As discussed.

**Review by insacuri — APPROVED** [review]
LGTM.
Comment for future devs:  Removing that PUBSUB_EMULATOR_HOST didn't seem to have any affect on the collectionExerciseEndpointIT test.  Both this branch and main looked like the errors in the test were similar (not being able to find certain subscriptions).  If it turns out it stopped working and I didn't notice, then it was probably this PR that did it.

**Pull request merged**